### PR TITLE
utopia88

### PR DESCRIPTION
**Pull request by kailundotwang**
Add PCB utopia88

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/utopia88/info.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                {"matrix":[5,0], "x":0, "y":5.25, "w":1.25}, 
+                {"matrix":[5,2], "x":1.25, "y":5.25, "w":1.25}, 
+                {"matrix":[5,3], "x":2.5, "y":5.25, "w":1.25}, 
+                {"matrix":[5,7], "x":4, "y":5.25, "w":7}, 
+                {"matrix":[5,11], "x":11.25, "y":5.25, "w":1.25}, 
+                {"matrix":[5,13], "x":12.5, "y":5.25, "w":1.25}, 
+                {"matrix":[5,14], "x":13.75, "y":5.25, "w":1.25}, 
+                {"matrix":[5,15], "x":15.25, "y":5.25}, 
+                {"matrix":[5,16], "x":16.25, "y":5.25}, 
+                {"matrix":[5,17], "x":17.25, "y":5.25}
```
Comment (dunk2k) [inline keyboards/utopia88/info.json:123]:
```suggestion
    },
    "community_layouts": ["tkl_f13_ansi_tsangan"],
    "layouts": {
        "LAYOUT_tkl_f13_ansi_tsangan": {
            "layout": [
                {"matrix":[0,0], "x":0, "y":0}, 
                {"matrix":[0,1], "x":1.25, "y":0}, 
                {"matrix":[0,2], "x":2.25, "y":0}, 
                {"matrix":[0,3], "x":3.25, "y":0}, 
                {"matrix":[0,4], "x":4.25, "y":0}, 
                {"matrix":[0,6], "x":5.5, "y":0}, 
                {"matrix":[0,7], "x":6.5, "y":0}, 
                {"matrix":[0,8], "x":7.5, "y":0}, 
                {"matrix":[0,9], "x":8.5, "y":0}, 
                {"matrix":[0,10], "x":9.75, "y":0}, 
                {"matrix":[0,11], "x":10.75, "y":0}, 
                {"matrix":[0,12], "x":11.75, "y":0}, 
                {"matrix":[0,13], "x":12.75, "y":0}, 
                {"matrix":[0,14], "x":14, "y":0}, 
                {"matrix":[0,15], "x":15.25, "y":0}, 
                {"matrix":[0,16], "x":16.25, "y":0}, 
                {"matrix":[0,17], "x":17.25, "y":0},
 
                {"matrix":[1,0], "x":0, "y":1.25}, 
                {"matrix":[1,1], "x":1, "y":1.25}, 
                {"matrix":[1,2], "x":2, "y":1.25}, 
                {"matrix":[1,3], "x":3, "y":1.25}, 
                {"matrix":[1,4], "x":4, "y":1.25}, 
                {"matrix":[1,5], "x":5, "y":1.25}, 
                {"matrix":[1,6], "x":6, "y":1.25}, 
                {"matrix":[1,7], "x":7, "y":1.25}, 
                {"matrix":[1,8], "x":8, "y":1.25}, 
                {"matrix":[1,9], "x":9, "y":1.25}, 
                {"matrix":[1,10], "x":10, "y":1.25}, 
                {"matrix":[1,11], "x":11, "y":1.25}, 
                {"matrix":[1,12], "x":12, "y":1.25}, 
                {"matrix":[1,14], "x":13, "y":1.25, "w":2}, 
                {"matrix":[1,15], "x":15.25, "y":1.25}, 
                {"matrix":[1,16], "x":16.25, "y":1.25}, 
                {"matrix":[1,17], "x":17.25, "y":1.25},
 
                {"matrix":[2,0], "x":0, "y":2.25, "w":1.5}, 
                {"matrix":[2,2], "x":1.5, "y":2.25}, 
                {"matrix":[2,3], "x":2.5, "y":2.25}, 
                {"matrix":[2,4], "x":3.5, "y":2.25}, 
                {"matrix":[2,5], "x":4.5, "y":2.25}, 
                {"matrix":[2,6], "x":5.5, "y":2.25}, 
                {"matrix":[2,7], "x":6.5, "y":2.25}, 
                {"matrix":[2,8], "x":7.5, "y":2.25}, 
                {"matrix":[2,9], "x":8.5, "y":2.25}, 
                {"matrix":[2,10], "x":9.5, "y":2.25}, 
                {"matrix":[2,11], "x":10.5, "y":2.25}, 
                {"matrix":[2,12], "x":11.5, "y":2.25}, 
                {"matrix":[2,13], "x":12.5, "y":2.25}, 
                {"matrix":[2,14], "x":13.5, "y":2.25, "w":1.5}, 
                {"matrix":[2,15], "x":15.25, "y":2.25}, 
                {"matrix":[2,16], "x":16.25, "y":2.25}, 
                {"matrix":[2,17], "x":17.25, "y":2.25}, 

                {"matrix":[3,0], "x":0, "y":3.25, "w":1.75}, 
                {"matrix":[3,2], "x":1.75, "y":3.25}, 
                {"matrix":[3,3], "x":2.75, "y":3.25}, 
                {"matrix":[3,4], "x":3.75, "y":3.25}, 
                {"matrix":[3,5], "x":4.75, "y":3.25}, 
                {"matrix":[3,6], "x":5.75, "y":3.25}, 
                {"matrix":[3,7], "x":6.75, "y":3.25}, 
                {"matrix":[3,8], "x":7.75, "y":3.25}, 
                {"matrix":[3,9], "x":8.75, "y":3.25}, 
                {"matrix":[3,10], "x":9.75, "y":3.25}, 
                {"matrix":[3,11], "x":10.75, "y":3.25}, 
                {"matrix":[3,12], "x":11.75, "y":3.25}, 
                {"matrix":[3,13], "x":12.75, "y":3.25, "w":2.25},
                {"matrix":[4,1], "x":0, "y":4.25, "w":2.25}, 
                {"matrix":[4,2], "x":2.25, "y":4.25}, 
                {"matrix":[4,3], "x":3.25, "y":4.25}, 
                {"matrix":[4,4], "x":4.25, "y":4.25}, 
                {"matrix":[4,5], "x":5.25, "y":4.25}, 
                {"matrix":[4,6], "x":6.25, "y":4.25}, 
                {"matrix":[4,7], "x":7.25, "y":4.25}, 
                {"matrix":[4,8], "x":8.25, "y":4.25}, 
                {"matrix":[4,9], "x":9.25, "y":4.25}, 
                {"matrix":[4,10], "x":10.25, "y":4.25}, 
                {"matrix":[4,11], "x":11.25, "y":4.25}, 
                {"matrix":[4,13], "x":12.25, "y":4.25, "w":2.75}, 
                {"matrix":[4,16], "x":16.25, "y":4.25},

                {"matrix":[5,0], "x":0, "y":5.25, "w":1.5}, 
                {"matrix":[5,2], "x":1.5, "y":5.25}, 
                {"matrix":[5,3], "x":2.5, "y":5.25, "w":1.5}, 
                {"matrix":[5,7], "x":4, "y":5.25, "w":7}, 
                {"matrix":[5,11], "x":11, "y":5.25, "w":1.5}, 
                {"matrix":[5,13], "x":12.5, "y":5.25}, 
                {"matrix":[5,14], "x":13.5, "y":5.25, "w":1.5}, 
                {"matrix":[5,15], "x":15.25, "y":5.25}, 
                {"matrix":[5,16], "x":16.25, "y":5.25}, 
                {"matrix":[5,17], "x":17.25, "y":5.25}
```

File: keyboards/utopia88/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,      KC_PSCR, KC_SCRL, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,    KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,            KC_COMM,  KC_DOT,  KC_SLSH,          KC_RSFT,             KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),   KC_RGUI, KC_RCTL,    KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
```
Comment (dunk2k) [inline keyboards/utopia88/keymaps/default/keymap.c:29]:
```suggestion
    [0] = LAYOUT_tkl_f13_ansi_tsangan(
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,      KC_PSCR, KC_SCRL, KC_PAUS,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,    KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,            KC_COMM,  KC_DOT,  KC_SLSH,          KC_RSFT,             KC_UP,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      MO(1),   KC_RGUI, KC_RCTL,    KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_tkl_f13_ansi_tsangan(

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/utopia88/info.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                {"matrix":[5,11], "x":11, "y":5.25, "w":1.5}, 
+                {"matrix":[5,13], "x":12.5, "y":5.25}, 
+                {"matrix":[5,14], "x":13.5, "y":5.25, "w":1.5}, 
+                {"matrix":[5,15], "x":15.25, "y":5.25}, 
+                {"matrix":[5,16], "x":16.25, "y":5.25}, 
+                {"matrix":[5,17], "x":17.25, "y":5.25}
+            ]
+        }
+    }
+}
```
Comment (dunk2k) [inline keyboards/utopia88/info.json:132]: remove whitespace
```suggestion
{
    "keyboard_name": "Utopia88",
    "manufacturer": "Kelwin_Studio",
    "maintainer": "Kelwin_Studio",
    "url": "",
    "usb": {
        "vid": "0x0088",
        "pid": "0x0888",
        "device_version": "0.0.1"
    },
    "matrix_pins": {
        "cols": ["E6", "F0", "F1", "F4", "F5", "F6", "F7", "C7", "C6", "B6", "B5", "B4", "D7", "D6", "D4", "B2", "B3", "B1"],
        "rows": ["B7", "D5", "D3", "D2", "D1", "D0"]
    },
    "diode_direction": "COL2ROW",
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
     "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": false,
        "nkro": true,
        "backlight": false,
        "rgblight": false,
        "rgb_matrix": false,
        "audio": true,
        "encoder": false
    },
    "build": {
        "lto": true
    },
    "community_layouts": ["tkl_f13_ansi_tsangan"],
    "layouts": {
        "LAYOUT_tkl_f13_ansi_tsangan": {
            "layout": [
                {"matrix":[0,0], "x":0, "y":0},
                {"matrix":[0,1], "x":1.25, "y":0},
                {"matrix":[0,2], "x":2.25, "y":0},
                {"matrix":[0,3], "x":3.25, "y":0},
                {"matrix":[0,4], "x":4.25, "y":0},
                {"matrix":[0,6], "x":5.5, "y":0},
                {"matrix":[0,7], "x":6.5, "y":0},
                {"matrix":[0,8], "x":7.5, "y":0},
                {"matrix":[0,9], "x":8.5, "y":0},
                {"matrix":[0,10], "x":9.75, "y":0},
                {"matrix":[0,11], "x":10.75, "y":0},
                {"matrix":[0,12], "x":11.75, "y":0},
                {"matrix":[0,13], "x":12.75, "y":0},
                {"matrix":[0,14], "x":14, "y":0},
                {"matrix":[0,15], "x":15.25, "y":0},
                {"matrix":[0,16], "x":16.25, "y":0},
                {"matrix":[0,17], "x":17.25, "y":0},
 
                {"matrix":[1,0], "x":0, "y":1.25},
                {"matrix":[1,1], "x":1, "y":1.25},
                {"matrix":[1,2], "x":2, "y":1.25},
                {"matrix":[1,3], "x":3, "y":1.25},
                {"matrix":[1,4], "x":4, "y":1.25},
                {"matrix":[1,5], "x":5, "y":1.25},
                {"matrix":[1,6], "x":6, "y":1.25},
                {"matrix":[1,7], "x":7, "y":1.25},
                {"matrix":[1,8], "x":8, "y":1.25},
                {"matrix":[1,9], "x":9, "y":1.25},
                {"matrix":[1,10], "x":10, "y":1.25},
                {"matrix":[1,11], "x":11, "y":1.25},
                {"matrix":[1,12], "x":12, "y":1.25},
                {"matrix":[1,14], "x":13, "y":1.25, "w":2},
                {"matrix":[1,15], "x":15.25, "y":1.25},
                {"matrix":[1,16], "x":16.25, "y":1.25},
                {"matrix":[1,17], "x":17.25, "y":1.25},
 
                {"matrix":[2,0], "x":0, "y":2.25, "w":1.5},
                {"matrix":[2,2], "x":1.5, "y":2.25},
                {"matrix":[2,3], "x":2.5, "y":2.25},
                {"matrix":[2,4], "x":3.5, "y":2.25},
                {"matrix":[2,5], "x":4.5, "y":2.25},
                {"matrix":[2,6], "x":5.5, "y":2.25},
                {"matrix":[2,7], "x":6.5, "y":2.25},
                {"matrix":[2,8], "x":7.5, "y":2.25},
                {"matrix":[2,9], "x":8.5, "y":2.25},
                {"matrix":[2,10], "x":9.5, "y":2.25},
                {"matrix":[2,11], "x":10.5, "y":2.25},
                {"matrix":[2,12], "x":11.5, "y":2.25},
                {"matrix":[2,13], "x":12.5, "y":2.25},
                {"matrix":[2,14], "x":13.5, "y":2.25, "w":1.5},
                {"matrix":[2,15], "x":15.25, "y":2.25},
                {"matrix":[2,16], "x":16.25, "y":2.25},
                {"matrix":[2,17], "x":17.25, "y":2.25},

                {"matrix":[3,0], "x":0, "y":3.25, "w":1.75},
                {"matrix":[3,2], "x":1.75, "y":3.25},
                {"matrix":[3,3], "x":2.75, "y":3.25},
                {"matrix":[3,4], "x":3.75, "y":3.25},
                {"matrix":[3,5], "x":4.75, "y":3.25},
                {"matrix":[3,6], "x":5.75, "y":3.25},
                {"matrix":[3,7], "x":6.75, "y":3.25},
                {"matrix":[3,8], "x":7.75, "y":3.25},
                {"matrix":[3,9], "x":8.75, "y":3.25},
                {"matrix":[3,10], "x":9.75, "y":3.25},
                {"matrix":[3,11], "x":10.75, "y":3.25},
                {"matrix":[3,12], "x":11.75, "y":3.25},
                {"matrix":[3,13], "x":12.75, "y":3.25, "w":2.25},
                {"matrix":[4,1], "x":0, "y":4.25, "w":2.25},
                {"matrix":[4,2], "x":2.25, "y":4.25},
                {"matrix":[4,3], "x":3.25, "y":4.25},
                {"matrix":[4,4], "x":4.25, "y":4.25},
                {"matrix":[4,5], "x":5.25, "y":4.25},
                {"matrix":[4,6], "x":6.25, "y":4.25},
                {"matrix":[4,7], "x":7.25, "y":4.25},
                {"matrix":[4,8], "x":8.25, "y":4.25},
                {"matrix":[4,9], "x":9.25, "y":4.25},
                {"matrix":[4,10], "x":10.25, "y":4.25},
                {"matrix":[4,11], "x":11.25, "y":4.25},
                {"matrix":[4,13], "x":12.25, "y":4.25, "w":2.75},
                {"matrix":[4,16], "x":16.25, "y":4.25},

                {"matrix":[5,0], "x":0, "y":5.25, "w":1.5},
                {"matrix":[5,2], "x":1.5, "y":5.25},
                {"matrix":[5,3], "x":2.5, "y":5.25, "w":1.5},
                {"matrix":[5,7], "x":4, "y":5.25, "w":7},
                {"matrix":[5,11], "x":11, "y":5.25, "w":1.5},
                {"matrix":[5,13], "x":12.5, "y":5.25},
                {"matrix":[5,14], "x":13.5, "y":5.25, "w":1.5},
                {"matrix":[5,15], "x":15.25, "y":5.25},
                {"matrix":[5,16], "x":16.25, "y":5.25},
                {"matrix":[5,17], "x":17.25, "y":5.25}
            ]
        }
    }
}
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/utopia88/readme.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Utopia88
+
+![Utopia88](https://i.imgur.com/pRKQIIHh.jpeg)
+
+
+* Keyboard Maintainer: [kelwin](https://github.com/Kelwin0313)
+* Hardware Supported: Utopia88 PCBs, atmega32u4
+* Hardware Availability: Discord kelwin4598 or Github(https://github.com/Kelwin0313)
```
Comment (dunk2k) [inline keyboards/utopia88/readme.md:8]: I'm not sure, by this Hardware Availability entry, if I have to contact Kelwin4598 for PCB files or that this person is a direct seller. Could you add some text to elaborate?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/utopia88/readme.md (hunk 4 above)
Comment (kailundotwang) [reply keyboards/utopia88/readme.md:8]: I'll modify it later. thank you.

**Pull request closed without merging**